src/test/java/uk/ac/ebi/eva/pipeline/io/writers/AnnotationMongoWriterTest.java

```diff
@@ -188,17 +188,27 @@ public void shouldWriteAllFieldsIntoMongoDbMultipleSetsAnnotations() throws Exce
     public void shouldWriteSubstitutionScoresIntoMongoDb() throws Exception {
         String databaseName = mongoRule.getRandomTemporaryDatabaseName();
 
-        Annotation annotation = new Annotation("X", 1, 10, "A", "T",
-                VEP_VERSION, VEP_CACHE_VERSION);
+        ScoreMongo siftScore = new ScoreMongo(0.02, "deleterious");
+        ScoreMongo polyphenScore = new ScoreMongo(0.846, "possibly_damaging");
 
-        Score siftScore = new Score(0.02, "deleterious");
-        Score polyphenScore = new Score(0.846, "possibly_damaging");
-
-        ConsequenceType consequenceType = new ConsequenceType();
+        ConsequenceTypeMongo consequenceType = new ConsequenceTypeMongo(null, null,
+            null, null, null, null, null, null,
+            null, null, null, null, null, null);
         consequenceType.setSift(siftScore);
         consequenceType.setPolyphen(polyphenScore);
 
-        annotation.addConsequenceType(consequenceType);
+        AnnotationMongo annotation = new AnnotationMongo(
+            AnnotationMongo.buildAnnotationId("X", 1, "A", "T",
+                VEP_VERSION, VEP_CACHE_VERSION),
+            "X",
+            1,
+            10,
+            VEP_VERSION,
+            VEP_CACHE_VERSION,
+            null,
+            Collections.singleton(consequenceType));
```

Comment:
Would moving this up, to where the Annotation instantiation originally was, have any impact?

Reply:
Yes, when I test with moving this up to where the Annotation was originally instantiated, it is not possible for me to assign `consequenceType` object to the Annotation object. This is because AnnotationMongo does not have a setter, and the `addConsequenceType` method is private as of now (accessible only via `concatenate` method. But `concatenate` method also expects an `annotation` object which has it's `consequenceType` field populated. So the only way to set `consequenceType` is via AnnotationMongo's constructor. So that's why it's instantiation now happens after consequenceType. So consequence type can be passed as parameter in the constructor of AnnotationMongo.